lib/webhdfs/client_v1.rb

```diff
@@ -60,31 +62,65 @@ def initialize(host='localhost', port=50070, username=nil, doas=nil, proxy_addre
       @kerberos = false
       @kerberos_keytab = nil
       @http_headers = http_headers
+
+      @renew_delegation_token_time_hr = renew_token_hr
+      @d_token = nil
+
+      @latest_time = Time.now
+
+    end
+
+    def is_update_time
+      @latest_time + (@renew_delegation_token_time_hr * 60 * 60) <= Time.now
+    end
+
+    def get_cached_delegation_token
+      begin
+        if @d_token.nil?
+          @d_token = delegation_token(@username)
+          @latest_time = Time.now
+        elsif is_update_time
+          renew_delegation_token(@d_token)
+          @latest_time = Time.now
+        end
+      rescue => e
+        raise WebHDFS::RequestFailedError, e.message
+      end
+
+      @d_token
     end
 
     # curl -i -X PUT "http://<HOST>:<PORT>/webhdfs/v1/<PATH>?op=CREATE
     #                 [&overwrite=<true|false>][&blocksize=<LONG>][&replication=<SHORT>]
     #                 [&permission=<OCTAL>][&buffersize=<INT>]"
+    #                 [&delegation=<DELEGATION_TOKEN>]"
     def create(path, body, options={})
       if @httpfs_mode
         options = options.merge({'data' => 'true'})
       end
+      unless @renew_delegation_token_time_hr.nil?
```

Comment:
It's better to use `if @renew_kerberos_delegation_token_time_hour` (it can't be `false`).